### ISSUE TITLE
General UI consistency issue: we should have conventions that are respected in all menu popups.

### ISSUE THREAD

**Issue by graysage1** [CleverRaven/Cataclysm-DDA#12112]
Right now we have what I believe to be a serious issue where each individual menu popup does whatever it likes, without any sense of game-wide conventions or standards.

Let's take for example the key that is supposed to close the popup.  This is so basic.  The game designers should settle on one or two keys, for example, ESC and 'q'.  Period.  From then on, any popup should be always closeable with ESC or 'q', end of story.  Right now we have windows that only close if you hit a number, or 'x' or 'q'.  Some windows respect 'ESC'.  Some don't.  It's insanity.  

But worst of all, bionic menu auto-closes whenever you perform any interaction with any bionic!  This can be highly undesirable.  If I want to toggle 5 bionics in a row, why do I have to reopen the window 5 times?  If there are people who never want to toggle more than one bionic at a time, there should be an option for them, but I think this should not be the default behavior.  The default should be a generic behavior that's consistent across all menus.

Another issue I have is tabbing.  Right now we use many multi-tab windows because the game is complex and this is both helpful and required in my view.  So we need tabbing conventions.

Why is it that when I use a fabrication menu, I can use PG-UP and PG-DOWN keys in addition to TAB to move across the tabs, but when I use the bionics menu, I can only use TAB and not PG-UP and PG-DOWN?  This should not be coded on a per-menu basis.  This should be a generic behavior for many reasons.

All this follows one theme: it appears different people are coding different menus however they please, without any sense of any kind of coherent vision of the game.  What is the game UI like overall?  Well, there seems to be no vision for it.  Instead every menu is a special snowflake with its own special behaviors.
##

**Comment by sparr** [comment]
Don't forget < and > in tabbed windows, sometimes! :/

**Comment by drbig** [comment]
(Note: this is for **General UI consistency**)

I think what we need first is a more comprehensive document describing all found inconsistencies. With such document it should be clear (or at least easier) to tell exactly how the UI should look and work overall. Then from this a TODO list for coding can be done, and a general official guideline. We do wan't to keep stuff consistent after we manage to fix what we have now.

Do you agree?

We could open a shared document for this on google docs/wherever.

**Comment by graysage1** [comment]
Would the core devs like to have a document as you describe?

**Comment by DavidKeaton** [comment]
yes. I was considering doing a ui update so all of the ui is under one hood.

**Comment by qbradq** [comment]
I agree a shared document needs to be compiled. The inconsistent ui is a barrier to entry for newcomers. And a have with this high of a learning curve needs as few of those as possible.

Could we perhaps use a wiki page for this?

**Comment by drbig** [comment]
I propose something much simpler and accessible: https://etherpad.mozilla.org/v82bPSjbcO

I'm also wondering about moving this discussion to a new issue solely dedicated to UI Inconsistency Hunting.

**Comment by graysage1** [comment]
@drbig Nice tool.

>I'm also wondering about moving this discussion to a new issue solely dedicated to UI Inconsistency Hunting.

Then what will be the purpose of this issue?

**Comment by drbig** [comment]
@graysage1 to be closed. @BevapDin's 'synchronized' basic popup 'escape' already.

**Comment by qbradq** [comment]
Can you help me see where @BevapDin level-set the escape behavior? Issue
#12378 was closed as a duplicate of this issue, not because of a PR.

**Comment by drbig** [comment]
@qbradq Indeed my bad. I assumed that #12378 was a PR... Sorry!

**Comment by DavidKeaton** [comment]
I will try to keep on eye on this as well @drbig, left a note about filtering on the page. Expect a comments/review on one soonish. :-)

**Comment by Leland** [comment]
This is extremely important. The list of related/duplicate issues should indicate that. I'll work on getting a list of current navigation buttons together.

**Comment by Leland** [comment]
See #19887 and #17469 and #19685

**Comment by Maleclypse** [comment]
I'm closing this as imgui is driving us towards a single convention where possible.
